### ISSUE TITLE
Explain what the heck are Maven Groups in the README

### ISSUE THREAD

**Issue by EnnuiL** [QuiltMC/quilt-example-mod#22]
While I do suggest the usage of the user's GitHub page as a replacement for one, there is one tiny issue:
 - Maven groups are never properly explained, confusing people in the process
 - The "use your GitHub page if you don't know what it is" trick may work, but it's not really explained, the association of it to Maven groups isn't strong, and it doesn't explain what are Maven groups
 - We need to explain Maven groups
 - That's all :P

**Comment by Cabbasca** [comment]
the fabric mod template doesn't explain what they are so neither should quilt

**Comment by Ecorous** [comment]
> the fabric mod template doesn't explain what they are so neither should quilt

That's effectively saying: fabric is bad so quilt shouldn't be better!!!!

**Comment by Cabbasca** [comment]
not really, it shouldn't be necessary to explain stuff in a template mod

**Comment by OroArmor** [comment]
If it's something we expect people to change it should be something we expect people to know what it is

**Comment by lukebemish** [comment]
I definitely agree, this is something the template should explain. I've seen too many mods in the wild using either (a) a completely nonsensical group, (b) a domain they don't own, or (c) something of no relation to the package name they use. Heck, I was personally very confused by group and package naming when I first did java/modding stuff, coming to it from outside the ecosystem. Explaining that the maven group name is similar to the package name prefix in terms of what you should use, and explaining options for what to use for both, is a great idea.
